include/wasmtime.hh

```diff
@@ -2483,6 +2503,16 @@ public:
     return results;
   }
 
+  TrapResult<std::vector<Val>> call(Store::Context cx,
+                                    const std::vector<Val> &params) const {
+    return this->call(cx, params.begin(), params.end());
+  }
+
+  TrapResult<std::vector<Val>> call(Store::Context cx,
+                                    const std::initializer_list<Val> &params) const {
+    return this->call(cx, params.begin(), params.end());
+  }
+
```

Comment:
I don't know what version of C++ is supported here (you might be able to leverage ranges), but generally I would expect a single templatized version like so:

```c++
  template<typename C>
  TrapResult<std::vector<Val>> call(Store::Context cx,
                                    C &&params) const {
    return this->call(cx, std::begin(params), std::end(params));
  }
```

Inspiration you can take is from Google's absl library container algorithms: https://github.com/abseil/abseil-cpp/blob/74df6975aef6d6fa2020313922e9a94b42364f38/absl/algorithm/container.h#L191-L200

That also makes it compatible C++20 ranges: https://en.cppreference.com/w/cpp/ranges

Which then you could constrain to a concept if needed:

```c++
  TrapResult<std::vector<Val>> call(Store::Context cx,
                                    std::ranges::input_range auto &&params) const {
    return this->call(cx, std::begin(params), std::end(params));
  }
```

Reply:
Oh nice, thanks!

Is there a way to influence via the `typename`/`template` parameter what the collection should contain? If I replaced the two `call` overloads added in this PR with your suggestion it fails for `{}` as it can't infer the type of the list and it also fails for `{1}` as it doesn't infer that it should be `Val` (I think)

Reply:
I don't think there is a way to do this without the initializer list overload. At least not with concepts: https://godbolt.org/z/c91sancY5. Maybe there is some SFINAE magic here somehow?

Reply:
Ah ok, no worries then! I'll leave it as-is and see if folks have issues in that case. (I barely know what SFINAE is myself other than that it's a thing)